packages/core/src/app/customer/FastllanePrivacySettings.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FunctionComponent, useEffect } from 'react';
+
+import { CheckoutContextProps } from '@bigcommerce/checkout/payment-integration-api';
+import { withCheckout } from '../checkout';
+import { PaymentMethod } from '@bigcommerce/checkout-sdk';
+
+export interface WithCheckoutProps {
+    providerWithCustomCheckout?: string;
+    getPaymentMethod: (methodId: string) => PaymentMethod | undefined;
+}
+
+interface FastlaneWatermarkOptions {
+    includeAdditionalInfo: boolean
+}
+
+interface FastlaneWatermarkComponent {
+    FastlaneWatermarkComponent: (options?: FastlaneWatermarkOptions) => Promise<FastlanePrivacySettings>
+}
+
+declare global {
+    interface Window {
+        paypalFastlane: FastlaneWatermarkComponent;
+        braintreeFastlane: FastlaneWatermarkComponent;
+    }
+}
+
+interface FastlanePrivacySettings {
+    render(container: string): void;
+}
+
+const FastlanePrivacySettings: FunctionComponent<WithCheckoutProps> = (props: WithCheckoutProps) => {
+    const {
+        getPaymentMethod,
+        providerWithCustomCheckout,
+    } = props;
+
+    useEffect(() => {
+        if (providerWithCustomCheckout === 'braintree' || providerWithCustomCheckout === 'braintreeacceleratedcheckout') {
+            const paymentMethod = getPaymentMethod(providerWithCustomCheckout);
+
+            if(paymentMethod?.initializationData?.isFastlanePrivacySettingEnabled) {
+                window.braintreeFastlane.FastlaneWatermarkComponent({
```

Comment:
Can we be so sure that we always have an `braintreeFastlane` in the `window`?

Reply:
if we have `paymentMethod?.initializationData?.isFastlanePrivacySettingEnabled` feature turned on we should have either paypalFastlane or braintreeFastlane property inside window that will be loaded before in fastlane customer strategy